### PR TITLE
testing - add memo field into payment transaction so parsePaymentMemo…

### PR DESCRIPTION
**Pull request by darkdarkdragon**
…s from parsePayment is tested

### CODE REVIEW

File: test/fixtures/api/responses/get-transactions.json [hunk 1]
```diff
@@ -3,8 +3,16 @@
     "type": "payment",
     "address": "r9cZA1mLK5R5Am25ArfXFmqgNwjZgnfk59",
     "sequence": 4,
-    "id": "4C37C92576DEB000D13B07F4D3F99F968BD86B6B83A840BEFFB2BFC8A042A81B",
+    "id": "99404A34E8170319521223A6C604AF48B9F1E3000C377E6141F9A1BF60B0B865",
     "specification": {
+      "memos": [
+        {
```
Comment (sublimator) [inline test/fixtures/api/responses/get-transactions.json]: In the context of RippleAPI carte blanche, is there much reason to expose these low level fields (`MemoType` / `MemoFormat`) vs something like `{"type"  : "...",  "format" : "...", "data" : "..."}` ?

Comment (sublimator) [inline test/fixtures/api/responses/get-transactions.json]: There seems to be a schema:  https://github.com/ripple/ripple-lib/blob/develop/src/api/common/schemas/memo.json

**Comment by sublimator** [comment]
Again, not specifically related to this pull request, but on the topic of memos, I notice that RippleAPI only seems to support memos for Payment transactions, when they actually work for any transaction: https://github.com/ripple/rippled/blob/develop/src/ripple/protocol/impl/TxFormats.cpp#L113

File: test/fixtures/api/responses/get-transactions.json (hunk 1 above)
Comment (clark800) [inline test/fixtures/api/responses/get-transactions.json]: Yeah, looks like memo parsing is not working yet. We should fix that before merging this pull request. @darkdarkdragon Can you fix parsePaymentMemos to return memos in the above schema format?

**Comment by clark800** [comment]
Supporting memos on non-payment transaction types is low-enough priority that we can leave that for later. Will make a note of it.

File: test/fixtures/api/responses/get-transactions.json (hunk 1 above)
Comment (darkdarkdragon) [inline test/fixtures/api/responses/get-transactions.json]: ok
BTW, as I understand, that one
https://github.com/ripple/ripple-client/blob/develop/src/js/tabs/send.controller.js#L1121
should be 
`tx.addMemo('client', 'plain/text', 'rt' + $scope.version);`

Comment (clark800) [inline test/fixtures/api/responses/get-transactions.json]: Looks like it

Comment (darkdarkdragon) [inline test/fixtures/api/responses/get-transactions.json]: done

**Comment by clark800** [comment]
LGTM

**Pull request merged**